### PR TITLE
ENH: S1 DA for noahmp401

### PR DESCRIPTION
**Pull request by mbechtold**
This PR adds all S1 DA modules (soil moisture only or soil moisture and LAI updating) that were already present in noahmp36  to noahmp401.

### CODE REVIEW

File: lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90 [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  count1 = 0.0
+  do i=1,LIS_rc%npatch(n,LIS_rc%lsm_index),LIS_rc%nensem(n)
+     do m=1,LIS_rc%nensem(n)
+        t = i+m-1
+        gid = LIS_surface(n,LIS_rc%lsm_index)%tile(t)%index
+        obs_pred(gid,m)= obs_pred(gid,m) + s0VV(t)
+        count1(gid,m) = count1(gid,m) +1
+     enddo
+  enddo
+
```
Comment (gdelannoy) [inline lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90:95]: (1) count1 for s0VV is never used, i.e. no average s0VV is computed? (2) The order of routines for VV and VH is different and not intuitive. (3) what is done with obs_pred_VV/VH?

File: lis/plugins/LIS_lsmda_pluginMod.F90 [hunk 2]
```diff
@@ -513,6 +528,8 @@ subroutine LIS_lsmda_plugin
    external noahmp401_scale_veg
    external noahmp401_descale_veg
    external noahmp401_veg_DAlog
+   external noahmp401_transform_veg
```
Comment (gdelannoy) [inline lis/plugins/LIS_lsmda_pluginMod.F90:531]: Not sure why this is added, does not seem to be used anywhere. Does no harm.

**Review by gdelannoy — CHANGES_REQUESTED** [review]
double check noahmp401_getSig0VVVHpred?

File: lis/plugins/LIS_lsmda_pluginMod.F90 (hunk 2 above)
Comment (mbechtold) [reply lis/plugins/LIS_lsmda_pluginMod.F90:531]: Indeed. Removed this, together with noahmp401_map_veg which is also not used anymore.

File: lis/surfacemodels/land/noahmp.4.0.1/da_soilmLAI/noahmp401_updatesoilmLAI.F90 [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+  call ESMF_AttributeGet(laiField,"Max Value",laimax,rc=status)
+  call LIS_verify(status,&
+       "ESMF_AttributeGet: laiField Max Value failed in noahmp401_updatesoilmLAI")
+  call ESMF_AttributeGet(laiField,"Min Value",laimin,rc=status)
+  call LIS_verify(status,&
+       "ESMF_AttributeGet: laiField Min Value failed in noahmp401_updatesoilmLAI")
+
+
+  do t=1,LIS_rc%npatch(n,LIS_rc%lsm_index)
```
[the remark this answers is not in the document]
Comment (mbechtold) [reply lis/surfacemodels/land/noahmp.4.0.1/da_soilmLAI/noahmp401_updatesoilmLAI.F90:151]: The *da_soilmLAI* routines used for the VVVH DA case in which soil moisture and vegetation are jointly updated were inherited from the two default DA modules *da_soilm* and *da_lai*. 
The physical bound check is done differently for those two modules.
For da_soilm, the Min and Max is checked *only* in da_soilm/noahmp401_qcsoilm.F90 (no check for physical bounds in the 'update' routine). For da_lai, the Min and Max is also checked in the update routine. We never deviated from the noahmp default procedures for the sm and sm-lai Sentinel-1 DA modules.

File: lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90 (hunk 1 above)
Comment (mbechtold) [reply lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90:95]: (1) count1 is used for s0VV and s0VH simultaneously at end of the subroutine. I added a comment for clarification. 
(2) correct, thanks for pointing this out. It is changed now. I also outcommented an unused section to keep it for possible later use.
(3) nothing, see (2)

**Comment by mbechtold** [comment]
Thanks, good catches of poor documentation / unclear coding. 
There was however no bug, zero-diff of results.

File: lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90 (hunk 1 above)
Comment (gdelannoy) [reply lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90:100]: looks better now!

**Review by gdelannoy — APPROVED** [review]
Looks good.

**Pull request merged**